font/src/ft/mod.rs

```diff
@@ -279,12 +347,15 @@ impl FreeTypeRasterizer {
                 }
             }
 
+            // Reuse the original fontkey if you're reloading the font
+            let key = if let Some(key) = key { key } else { FontKey::next() };
+
             let face = Face {
                 ft_face,
-                key: FontKey::next(),
-                load_flags: Self::ft_load_flags(pattern),
-                render_mode: Self::ft_render_mode(pattern),
-                lcd_filter: Self::ft_lcd_filter(pattern),
+                key,
+                load_flags: Self::ft_load_flags(&pattern),
+                render_mode: Self::ft_render_mode(&pattern),
+                lcd_filter: Self::ft_lcd_filter(&pattern),
```

Comment:
None of these need to be references, since `pattern` is already a `&PatternRef`.

Reply:
I really wish there would be warnings for such things.